qoqo-iqm/tests/integration/backend.rs

```diff
@@ -24,49 +24,43 @@ fn test_creating_backend_deneb_device() {
     // Test if Backend is created successfully with a dummy access token
     Python::with_gil(|py| {
         // get Python type (i.e. Python class) corresponding to DenebDeviceWrapper Rust type
-        let device_type = py.get_type::<devices::DenebDeviceWrapper>();
+        let device_type = py.get_type_bound::<devices::DenebDeviceWrapper>();
         let device = device_type
             // Instantiate Python class
             .call0()
-            .unwrap()
-            .downcast::<PyCell<devices::DenebDeviceWrapper>>()
             .unwrap();
-        let backend_type = py.get_type::<BackendWrapper>();
+        let backend_type = py.get_type_bound::<BackendWrapper>();
         let _backend = backend_type
-            .call1((device, "DUMMY_ACCESS_TOKEN"))
+            .call1((
+                device.downcast::<devices::DenebDeviceWrapper>().unwrap(),
+                "DUMMY_ACCESS_TOKEN",
+            ))
             .unwrap()
-            .downcast::<PyCell<BackendWrapper>>()
+            .downcast::<BackendWrapper>()
             .unwrap();
     });
 
     if env::var("IQM_TOKEN").is_ok() {
         // Test if Backend correctly retrieves access token from environment variable
         Python::with_gil(|py| {
-            let device_type = py.get_type::<devices::DenebDeviceWrapper>();
-            let device = device_type
-                .call0()
-                .unwrap()
-                .downcast::<PyCell<devices::DenebDeviceWrapper>>()
-                .unwrap();
-            let backend_type = py.get_type::<BackendWrapper>();
+            let device_type = py.get_type_bound::<devices::DenebDeviceWrapper>();
+            let device = device_type.call0().unwrap();
+            let backend_type = py.get_type_bound::<BackendWrapper>();
             let _backend = backend_type
-                .call1((device,))
+                .call1((device.downcast::<devices::DenebDeviceWrapper>().unwrap(),))
```

Comment:
What's the reason for moving the downcast inside this function call? It looks more neat to me to leave it chained after the `call0` call

Reply:
If we leave it chained we get the following error: `temporary value dropped while borrowed
creates a temporary value which is freed while still in use` 
I think because now the device_type is a `Bound<PyType>` and not just a `PyType` the `call0` will remove the bound, convert it to PyAny and recreate a Bound which is a memory allocation. If we chain it the variable will be temporary so it will be freed at the end of the chain and `downcast` that borrows it will be using something that was freed. So we need to have the result of `call0` assigned to a variable in order for `downcast` to be able to borrow it.
A clearer equivalent would be
`let s = "test".to_owned().as_str();`
Where `to_owned` is `call0` and `as_str` is `downcast`

Reply:
I see, very good